src/tracer.cc

```diff
@@ -287,24 +287,12 @@ void
 TraceManager::Trace::CaptureTimestamp(
     const std::string& name, uint64_t timestamp_ns)
 {
-  if (setting_->level_ & TRITONSERVER_TRACE_LEVEL_TIMESTAMPS) {
-    std::lock_guard<std::mutex> lk(mtx_);
-    std::stringstream* ss = nullptr;
-    {
-      if (streams_.find(trace_id_) == streams_.end()) {
-        std::unique_ptr<std::stringstream> stream(new std::stringstream());
-        ss = stream.get();
-        streams_.emplace(trace_id_, std::move(stream));
-      } else {
-        ss = streams_[trace_id_].get();
-        // If the string stream is not newly created, add "," as there is
-        // already content in the string stream
-        *ss << ",";
-      }
-    }
-    *ss << "{\"id\":" << trace_id_ << ",\"timestamps\":["
-        << "{\"name\":\"" << name << "\",\"ns\":" << timestamp_ns << "}]}";
-  }
+  LOG_TRITONSERVER_ERROR(
+      TRITONSERVER_InferenceTraceSetName(trace_, name.c_str()),
+      "set trace name");
+  LOG_TRITONSERVER_ERROR(
+      TRITONSERVER_InferenceTraceReportActivity(trace_, timestamp_ns),
+      "report trace activity");
```

Comment:
I don't think this function requires to go through the C API for tracing, it is the helper function to be use by the frontend (which already knows the concrete instance of the trace manager), if the reason of change is to reuse code, it should be able to call the `TraceActivity()` instead